src/utilities/validation.ts

```diff
@@ -80,3 +79,44 @@ export function validateDateWithInterval(
   }
   return { msg: "", err: false };
 }
+
+export function validateConfirmPassword(password: string, confirmPassword: string): validation {
+  if (
+    password.length < CHAR_LIMIT.MIN_PASSWORD ||
+    confirmPassword.length < CHAR_LIMIT.MIN_PASSWORD
+  ) {
+    return {
+      msg: `ช่องนี้ต้องมีตัวอักษรอย่างน้อย ${CHAR_LIMIT.MIN_PASSWORD} ตัว`,
+      err: true,
+    };
+  }
+  if (password !== confirmPassword) {
+    return {
+      msg: "Password และ Confirm Password ต้องเหมือนกัน",
+      err: true,
+    };
+  }
+  return { msg: "", err: false };
+}
+
+export function validateConfirmNewPassword(
```

Comment:
I found that if you enter only space equally and 6+ times in both `New Password` and `Confirm New Password` field (for example `      `, it would be a valid password. Personally, I think it quite not right to have this password. Could you please check this kind of password that it should not be include space in it (both only space and letter mixing with space)

Reply:
May be error message would be "รหัสผ่านไม่สามารถประกอบด้วยช่องว่าง".

Reply:
Got it 😸

Reply:
Solved. Pls check

Reply:
I think password containing space should be check before password min char since when you type space, it would be appear like you type password when you are not show password which will confuse user, like, it seems you typing more than 6 but the error tell you to type at least 6 (pic below is when i type `         ` both in `New Password` and `Confirm New Password`). Seems like now error message "รหัสผ่านไม่สามารถประกอบด้วยช่องว่าง" only occurs when there are space between char.
<img width="1530" alt="Screenshot 2566-02-25 at 22 51 51" src="https://user-images.githubusercontent.com/62459518/221366693-24205641-483e-400c-91c6-8828ae4de5b9.png">

Reply:
For me, it's like
1. check empty
2. check contain space (starting, between, ending)
3. check min char
4. check match
feel free to discuss ka

Reply:
I would go on your priority with higher priority on "new password"'s error higher priority which means the "confirmm new password"'s error wont display unless "new password" is valid. What do you think?

Reply:
`naughty action`

Reply:
Sounds like a good idea ka.

Reply:
Solved. Pls check 🥺

Reply:
1. Seems like when new password is all space the error still be min char error.  I guess may be you should swap line 20-25 with line 27-29 in `validation.ts` ? or else may be it woule be, 
```// check contain space (starting, between, ending)
  if (password.includes(" ")) {
    return { msg: "รหัสผ่านไม่สามารถประกอบด้วยช่องว่าง", err: true };
  }
  // check min char
  if (password.trim().length < CHAR_LIMIT.MIN_PASSWORD || password.trim().length == 0) {
    return {
      msg: `รหัสผ่านต้องมีตัวอักษรอย่างน้อย ${CHAR_LIMIT.MIN_PASSWORD} ตัวอักษร`,
      err: true,
    };

Reply:
2. from your new code, since `confirm new password`'s error wont display unless new password is valid, is it good idea if when `new password` is valid, whatever `confirm new password` is just check only it matches with `new password` or not so the error would show only password not match? This idea came through my mind when I test it with new password is valid but confirm new password contain space and it showed that password shouldn't contain space and it seems a little bit weird to me. What do you think?

Reply:
1. so `      ` !== empty, right?
2. I'm ok with both. If you say so, I would change it.
If there are further recommendation in your mind pls let me know. 😄

Reply:
1. yes, in this case what I mean by empty is not containing anything (the field is still show place holder text in this page) and containing space means having space in any order, for example, `       `, `   111`, `111   `, `1    1`. I'm sorry if my words made you confuse a lot.
2. If you agree, please make change due to my recommendation. I really do appreciate it. Also, what I mean in the above comment for `confirm new password` is something like this (hope you see a bit clearer by my comment), feels free do change anything ka. This is just my recommendation (also opinion) ka :relaxed:
```
export function validateConfirmNewPassword(
  newPassword: string,
  confirmNewPassword: string
): validation {
  const newPasswordErr: validation = validatePassword(newPassword);

  if (newPasswordErr.err) {
    return newPasswordErr;
  }
  if (newPassword !== confirmNewPassword) { // if confirmNewPassword is invalid, it wouldn't match newPassword which is valid anyway
    return {
      msg: "New Password และ Confirm New Password ต้องเหมือนกัน",
      err: true,
    };
  }
  return { msg: "", err: false };
}
```

Reply:
Solved krub

Reply:
Looks good overall na but I just have a little curiosity about why you separately check `password.length == 0` and `password.trim().length < CHAR_LIMIT.MIN_PASSWORD` since the error message is the same (like `password.length == 0 || `password.trim().length < CHAR_LIMIT.MIN_PASSWORD`) ? Are there any reasons behind it (priority?) ?

Reply:
yes, mainly about priority. To merge them, there are 2 possible output. Firstly, check length before check space. This priority case is check empty + check min > check space. In another word, `11 11` is min char error which isn't suppose to be. On the other hand,  check empty after check space leads to on space error on `    ` which also isn't suppose to be.

Reply:
yes, priority. I've come up with idea to merge them let me try.

Reply:
Already delete duplicated code and seem like it works properly.